spark/src/main/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuery.scala

```diff
@@ -60,37 +127,199 @@ trait OptimizeMetadataOnlyDeltaQuery {
     Some(numRecords)
   }
 
-  /** e.g. sql("SELECT COUNT(*) FROM <deta-table>").collect() */
-  object CountStarDeltaTable {
-    def unapply(plan: Aggregate): Option[Long] = plan match {
-      case Aggregate(
-          Nil,
-          Seq(Alias(AggregateExpression(Count(Seq(Literal(1, _))), Complete, false, None, _), _)),
-          PhysicalOperation(_, Nil, DeltaTable(i: TahoeLogFileIndex))) if i.partitionFilters.isEmpty
-        => extractGlobalCount(i)
-      case _ => None
+  val columnStatsSupportedDataTypes: HashSet[DataType] = HashSet(
+    ByteType,
+    ShortType,
+    IntegerType,
+    LongType,
+    FloatType,
+    DoubleType,
+    DateType)
+
+  case class DeltaColumnStat(
+    min: Any,
+    max: Any)
+
+  def extractGlobalColumnStats(tahoeLogFileIndex: TahoeLogFileIndex):
```

Comment:
nit: private

Reply:
Please also add some docs.

Reply:
This is a relatively large method and has too much logic in it. Please split this method into smaller methods each one with clear functionality.